sql/core/src/test/scala/org/apache/spark/sql/execution/QueryExecutionSuite.scala

```diff
@@ -336,6 +336,18 @@ class QueryExecutionSuite extends SharedSparkSession {
     }
   }
 
+  test("SPARK-47033: EXECUTE IMMEDIATE USING does not recognize session variable names") {
+    spark.sql("DECLARE parm = 'Hello';")
+
+    val originalQuery = spark.sql("EXECUTE IMMEDIATE 'SELECT :parm' USING parm AS parm;")
+    val newQuery = spark.sql("EXECUTE IMMEDIATE 'SELECT :parm' USING parm;")
```

Comment:
can we also test `USING system.session.parm`? I think the alias should still be `parm`

Reply:
Like `spark.sql("EXECUTE IMMEDIATE 'SELECT :parm' USING system.session.parm AS parm;")`
or
`spark.sql("EXECUTE IMMEDIATE 'SELECT :parm' USING system.session.parm;")`

Reply:
The second query does not work, but I guess this is meant to fail. Engine should not predict if the user wanted to use `system.session.parm` as just `parm` without using an alias. First one passes.

Reply:
we should follow the same alias generation rule of the SELECT statement. For `SELECT system.session.parm`, the final output column name is `parm`, because we use `UnresolvedAttribute.nameParts.last` as the alias name.

Reply:
Agree. But should it be matched with `:parm`? The USING statement does not know if `:parm` is `system.session.parm`.

Reply:
Should USING statement match `:parm` with `system.session.parm`? That would require manually removing the parts of the name.

Reply:
It matches the SELECT statement behavior so I'm fine with it. cc @srielau to confirm

Reply:
Yes, that's what we want. Any column like reference works that way: column alias, column, function parameter (which can be qualified by function name), variable, function